### PR TITLE
Added domainMapping tests

### PR DESCRIPTION
**Pull request by liadlauber**

### CODE REVIEW

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		}
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, domainMapping)
+		}, TimeoutCapp, CappCreationInterval).Should(BeTrue(), "Should find a resource.")
+	})
+
+	It("Should create DomainMapping with secret when Creating an https Capp instance", func() {
+		By("Creating a secret")
+		secretName := utilst.GenerateSecretName()
+		secret := v1.Secret{
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:36]: you can transfer the object creation to the mocks and pass it name and namespace

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+var _ = Describe("Validate DomainMapping functionality", func() {
+
+	It("Should create DomainMapping when Creating a Capp instance", func() {
+		By("Creating a capp with a route")
+		routeCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		routeCapp.Spec.RouteSpec.Hostname = routeHostname
+		utilst.CreateCapp(k8sClient, routeCapp)
+		By("Checking if the domainMapping was created successfully")
+		domainMapping := &knativev1beta1.DomainMapping{
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:22]: you can transfer the object creation to the mocks and pass it name and namespace

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+var _ = Describe("Validate DomainMapping functionality", func() {
+
+	It("Should create DomainMapping when Creating a Capp instance", func() {
+		By("Creating a capp with a route")
+		routeCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		routeCapp.Spec.RouteSpec.Hostname = routeHostname
+		utilst.CreateCapp(k8sClient, routeCapp)
+		By("Checking if the domainMapping was created successfully")
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:21]: divide `By` with blank line

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      routeHostname,
+				Namespace: routeCapp.Namespace,
+			},
+		}
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, domainMapping)
+		}, TimeoutCapp, CappCreationInterval).Should(BeTrue(), "Should find a resource.")
+		CreatedDomainMapping := utilst.GetDomainMapping(k8sClient, routeHostname, routeCapp.Namespace)
+		By("Deleting the capp instance")
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:77]: I don't think  you should separate the base `CRUD` logic to different `Its` because you repeating the creation of Capp and the profit of concurrency is minimal. The edge cases should be in different  `Its`.

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		utilst.CreateCapp(k8sClient, httpsCapp)
+		By("Checking if the secret reference exists at the domainMapping")
+		Eventually(func() string {
+			domainMapping := utilst.GetDomainMapping(k8sClient, routeHostname, httpsCapp.Namespace)
+			return domainMapping.Spec.TLS.SecretName
+		}, TimeoutCapp, CappCreationInterval).Should(Equal(secretName))
+	})
+
+	It("Should delete DomainMapping when deleting a Capp instance", func() {
+		By("Creating a capp with a route")
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:61]: Divide `Bys`

File: test/k8s_tests/domain_mapping_e2e_test.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+			return utilst.DoesResourceExist(k8sClient, createdCapp)
+		}, TimeoutCapp, CappCreationInterval).ShouldNot(BeTrue(), "Should not find a resource.")
+		By("Checking if the domainMapping was deleted successfully")
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, CreatedDomainMapping)
+		}, TimeoutCapp, CappCreationInterval).ShouldNot(BeTrue(), "Should not find a resource.")
+	})
+
+	It("Should update DomainMapping when updating route hostname", func() {
+		By("Creating a capp with a route")
```
Comment (fran2810) [inline test/k8s_tests/domain_mapping_e2e_test.go:89]: Divide `Bys` and unite with basic CRUD `It`

**Review by fran2810 — COMMENTED** [review]
Overall very nice. few minor fixes for improve readability. You have a bright future in QA automation 😄

**Comment by liadlauber** [comment]
@fran2810 

Thank you for the positive feedback! Your opinion means a lot to me, and I'm excited about the prospect of a future in QA automation!  🚀

**Review by fran2810 — APPROVED** [review]
/lgtm
